src/map.rs

```diff
@@ -2194,18 +2166,19 @@ where
                                 //
                                 // here are the possible cases:
                                 //
-                                //  - another thread already has a reference to the old value.
-                                //    they must have read it before the call to store_bin.
-                                //    because of this, that thread must be pinned to an epoch <=
-                                //    the epoch of our guard. since the garbage is placed in our
-                                //    epoch, it won't be freed until the _next_ epoch, at which
-                                //    point, that thread must have dropped its guard, and with it,
-                                //    any reference to the value.
+                                //  - another thread already has a reference to now_garbage.
+                                //    they must have read it before the call to swap while
+                                //    marked as active (holding a guard), and are included in
+                                //    the reference count. therefore t won't be freed until _after_
```

Comment:
we really do have a _lot_ of instances of this same argument :p

Reply:
Yeah :smile: